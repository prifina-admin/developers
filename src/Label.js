import React, { forwardRef } from "react";

import styled, { css } from "styled-components";

import { layout, space, typography, color, compose } from "styled-system";
import { default as styledProps } from "@styled-system/prop-types";
import PropTypes from "prop-types";

const systemProps = compose(layout, space, typography, color);

const nowrap = props => {
  return props.nowrap
    ? {
        whiteSpace: "nowrap",
      }
    : null;
};
const pointer = props => {
  return props.pointer ? { cursor: "pointer" } : null;
};
const hidden = props => {
  return props.hidden ? { visibility: "hidden" } : null;
};

/*
const VisuallyHidden = styled(Box)`
  border: 0px;
  clip: rect(0px, 0px, 0px, 0px);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0px;
  overflow: hidden;
  white-space: nowrap;
  position: absolute;
`;
*/

const LabelTheme = css`
  font-size: ${props => props.theme.textStyles[props.textStyle].fontSize};
  letter-spacing: ${props =>
    props.theme.textStyles[props.textStyle].letterSpacing};
  display: ${props => props.variation};
  width: 100%;
  margin: 0;
  color: ${props => props.theme.textStyles[props.textStyle].color};
  font-weight: ${props => props.theme.textStyles[props.textStyle].fontWeight};
  line-height: ${props => props.theme.textStyles[props.textStyle].lineHeight};
  ${systemProps};
  ${nowrap}
  ${pointer}
  ${hidden}
`;

const LabelElement = styled("label")`
  ${LabelTheme}
`;

const Label = forwardRef((props, ref) => {
  return <LabelElement {...props} ref={ref} />;
});
Label.propTypes = {
  ...styledProps.space,
  ...styledProps.layout,
  ...styledProps.typography,

  variation: PropTypes.oneOf(["block", "inline", "none"]),
  pointer: PropTypes.bool,
};
Label.defaultProps = {
  textStyle: "body",
};

Label.displayName = "Label";
Label.isLabel = true;

export default Label;
