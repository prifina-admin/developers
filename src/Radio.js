import React, {forwardRef} from "react";
import styled, { css } from "styled-components";
import { space, layout, compose } from "styled-system";
import PropTypes from "prop-types";

const systemProps = compose(layout, space);

const radioTheme = css`
  appearance: radio;
  cursor: pointer;
  border-radius: 50%;
  margin: 3px 3px 0px 5px;
  padding: initial;
  border: initial;
  
  ${systemProps}
`;

const RadioElement = styled.input`
  ${radioTheme}
`;

const Radio = forwardRef((props, ref) => {
   return <RadioElement {...props} ref={ref} />;
});

Radio.defaultProps = {
    as: "input",
    type: "radio",
    onChange: () => {},
};

Radio.propTypes = {
    onChange: PropTypes.func,
}

Radio.displayName = "Radio";

export default Radio;
