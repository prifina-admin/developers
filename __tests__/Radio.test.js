import React, {createRef} from "react";
import renderer from "react-test-renderer";
import ThemeProvider from "../src/theme/ThemeProvider";
import Label from "../src/Label";
import Radio from "../src/Radio";
import styled from "styled-components";
import {action} from "@storybook/addon-actions";

const LabelText = styled.span`
  vertical-align: middle;
  margin-left: 8px;
`;

const checkAction = e => {
    console.log("CLICK ", e, e.target);
    action(`${e.target.id} was clicked`)(e.target.value, e.target.checked);
};

test("Link renders correctly", () => {
    const tree = renderer.create(
        <ThemeProvider>
            <Label fontSize="14px" onClick={checkAction}>
                <Radio checked value="TABLE" />
                <LabelText>selected</LabelText>
            </Label>
            <Label fontSize="14px" onClick={checkAction}>
                <Radio value="OBJECT" />
                <LabelText>not selected</LabelText>
            </Label>
            <Label fontSize="14px">
                <Radio disabled />
                <LabelText>disabled</LabelText>
            </Label>
        </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

