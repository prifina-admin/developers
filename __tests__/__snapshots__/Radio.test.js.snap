// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <label
    className="Label__LabelElement-u5wfo1-0 grFNee"
    fontSize="14px"
    onClick={[Function]}
  >
    <div
      checked={true}
      className="Radio__RadioWrap-sc-1tv8lnt-0 dGosSt"
    >
      <input
        checked={true}
        className="Radio__RadioInput-sc-1tv8lnt-1 jUgNmh"
        type="radio"
        value="TABLE"
      />
      <svg
        className="Icon-y3r77h-0 hfhfoT"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M12 20a8 8 0 0 1-8-8a8 8 0 0 1 8-8a8 8 0 0 1 8 8a8 8 0 0 1-8 8m0-18A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2m0 5a5 5 0 0 0-5 5a5 5 0 0 0 5 5a5 5 0 0 0 5-5a5 5 0 0 0-5-5z\\" fill=\\"currentColor\\"/>",
          }
        }
        focusable={false}
        height="1.5rem"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1.5rem"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
    <span
      className="Radiotest__LabelText-sc-117ifwp-0 kdrKBa"
    >
      selected
    </span>
  </label>
  <label
    className="Label__LabelElement-u5wfo1-0 grFNee"
    fontSize="14px"
    onClick={[Function]}
  >
    <div
      className="Radio__RadioWrap-sc-1tv8lnt-0 dGosSt"
    >
      <input
        className="Radio__RadioInput-sc-1tv8lnt-1 jUgNmh"
        type="radio"
        value="OBJECT"
      />
      <svg
        className="Icon-y3r77h-0 hfhfoT"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M12 20a8 8 0 0 1-8-8a8 8 0 0 1 8-8a8 8 0 0 1 8 8a8 8 0 0 1-8 8m0-18A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z\\" fill=\\"currentColor\\"/>",
          }
        }
        focusable={false}
        height="1.5rem"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1.5rem"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
    <span
      className="Radiotest__LabelText-sc-117ifwp-0 kdrKBa"
    >
      not selected
    </span>
  </label>
  <label
    className="Label__LabelElement-u5wfo1-0 grFNee"
    fontSize="14px"
  >
    <div
      className="Radio__RadioWrap-sc-1tv8lnt-0 dGosSt"
      disabled={true}
    >
      <input
        className="Radio__RadioInput-sc-1tv8lnt-1 jUgNmh"
        disabled={true}
        type="radio"
      />
      <svg
        className="Icon-y3r77h-0 hfhfoT"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M12 20a8 8 0 0 1-8-8a8 8 0 0 1 8-8a8 8 0 0 1 8 8a8 8 0 0 1-8 8m0-18A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2z\\" fill=\\"currentColor\\"/>",
          }
        }
        focusable={false}
        height="1.5rem"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1.5rem"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
    <span
      className="Radiotest__LabelText-sc-117ifwp-0 kdrKBa"
    >
      disabled
    </span>
  </label>
</div>
`;
